com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/JdtSourceLookUpProvider.java

```diff
@@ -276,8 +278,14 @@ private CompilationUnit asCompilationUnit(String uri) {
              * setEnvironment(String [], String [], String [], boolean)
              * and a unit name setUnitName(String).
              */
-            parser.setEnvironment(new String[0], new String[0], null, true);
-            parser.setUnitName(Paths.get(filePath).getFileName().toString());
+            IFile resource = (IFile) JDTUtils.findResource(JDTUtils.toURI(uri),
+                    ResourcesPlugin.getWorkspace().getRoot()::findFilesForLocationURI);
+            if (resource != null && JdtUtils.isJavaProject(resource.getProject())) {
+                parser.setProject(JavaCore.create(resource.getProject()));
+            } else {
+                parser.setEnvironment(new String[0], new String[0], null, true);
+                parser.setUnitName(Paths.get(filePath).getFileName().toString());
```

Comment:
`parser.setUnitName(Paths.get(filePath).getFileName().toString());` is required for the case of Java project configured as well.

Reply:
There is NPE without the unit name:
```
!MESSAGE [error response][setBreakpoints]: Cannot invoke "org.eclipse.jdt.core.dom.ITypeBinding.getBinaryName()" because the return value of "org.eclipse.jdt.core.dom.AbstractTypeDeclaration.resolveBinding()" is null
!STACK 0
java.lang.NullPointerException: Cannot invoke "org.eclipse.jdt.core.dom.ITypeBinding.getBinaryName()" because the return value of "org.eclipse.jdt.core.dom.AbstractTypeDeclaration.resolveBinding()" is null
	at com.microsoft.java.debug.LambdaExpressionLocator.getFullyQualifiedTypeName(LambdaExpressionLocator.java:96)
	at com.microsoft.java.debug.plugin.internal.JdtSourceLookUpProvider.getBreakpointLocations(JdtSourceLookUpProvider.java:177)
	at com.microsoft.java.debug.core.adapter.handler.SetBreakpointsRequestHandler.convertClientBreakpointsToDebugger(SetBreakpointsRequestHandler.java:312)
	at com.microsoft.java.debug.core.adapter.handler.SetBreakpointsRequestHandler.handle(SetBreakpointsRequestHandler.java:108)
```

Reply:
Fixed